nodes/logic/evolver.py

```diff
@@ -0,0 +1,452 @@
+        print("evolver was %s seconds processing" % (time.time() - self.time_start))
+
+    def goal_achieved(self, fittest):
+        if self.mode == "MAX":
+            return fittest > self.fitness_goal
+        if self.mode == "MIN":
+            return fittest < self.fitness_goal
+
+    def store_data(self, population_all, fitness_all):
+        evolver_mem[self.node_id]["population_all"] = population_all
+        evolver_mem[self.node_id]["fitness_all"] = fitness_all
+        evolver_mem[self.node_id]["genes"] = self.genes
+        evolver_mem[self.node_id]["population"] = population_all[-1]
+        evolver_mem[self.node_id]["fitness"] = fitness_all[-1]
+
+    def evolve(self, update_list):
+        population_all = []
+        fitness_all = []
+        info = "Evolver Runned"
+        goal_achieved = False
+        for i in range(self.iterations - 1):
+            self.evaluate_fitness_g(update_list)
+            self.population_g.sort(key=lambda x: x.fitness, reverse=(self.mode == "MAX"))
+            population_all.append(self.population_genes())
+            actual_population_fitenss = self.population_fitness()
+            fitness_all.append(actual_population_fitenss)
+
+            if self.use_fitness_goal and self.goal_achieved(actual_population_fitenss[0]):
+                goal_achieved = True
+                info = "goal achieved in %s iterations" % (i + 1)
+                print(info)
+                break
+
+            self.print_time_info(i)
+            if (time.time() - self.time_start) > self.max_time:
+                info = "Max. time reached in %s iterations" % (i + 1)
+                print(info)
+                break
+
+            self.population_g = self.get_new_population(actual_population_fitenss)
+
+
+        if not goal_achieved:
+            self.evaluate_fitness_g(update_list)
+            self.population_g.sort(key=lambda x: x.fitness, reverse=(self.mode == "MAX"))
+            population_all.append(self.population_genes())
+            fitness_all.append(self.population_fitness())
+
+        self.store_data(population_all, fitness_all)
+        self.node.info_label = info
+
+
+class SvEvolverRun(bpy.types.Operator):
+
+    bl_idname = "node.evolver_run"
+    bl_label = "Evolver Run"
+
+    idtree: bpy.props.StringProperty(default='')
+    idname: bpy.props.StringProperty(default='')
+
+    def execute(self, context):
+        tree = bpy.data.node_groups[self.idtree]
+        node = bpy.data.node_groups[self.idtree].nodes[self.idname]
+
+        if not node.inputs[0].is_linked:
+            node.info_label = "Stopped - Fitness not linked"
+            return {'FINISHED'}
+
+        genotype_frame = node.genotype
+        evolver_mem[node.node_id] = {}
+
+        seed_set(node.r_seed)
+        np.random.seed(node.r_seed)
+
+        genes = get_genes(tree, genotype_frame)
+        update_list = make_tree_from_nodes([g.name for g in genes], tree)
+        population = Population(genes, node, tree)
```

Comment:
I see this like this:
```suggestion
        population = Population.get_genes(node, tree, genotype_frame)
```

Reply:
as there is no other call to the constructor I added the get_genes to the init function so it is 
`population = Population(node, tree, genotype_frame)`